src/compiler/visitorPublic.ts

```diff
@@ -377,7 +377,7 @@ function visitArrayWorker(
  * Starts a new lexical environment and visits a statement list, ending the lexical environment
  * and merging hoisted declarations upon completion.
  */
-export function visitLexicalEnvironment(statements: NodeArray<Statement>, visitor: Visitor, context: TransformationContext, start?: number, ensureUseStrict?: boolean, nodesVisitor: NodesVisitor = visitNodes) {
+export function visitLexicalEnvironment(statements: NodeArray<Statement>, visitor: Visitor, context: CoreTransformationContext, start?: number, ensureUseStrict?: boolean, nodesVisitor: NodesVisitor = visitNodes) {

```

Comment:
Hey @rbuckton, @jakebailey, we have some further changes that depend on these, could you please take a look?

Reply:
I don't think anything actually uses `CoreTransformationContext` anymore (`chainBundle` does, but currently its only ever used in places where a full transformation context is available). I'd originally added it as part of the node factory API refactor in https://github.com/microsoft/TypeScript/pull/35282, with the intent to leverage it in https://github.com/microsoft/TypeScript/pull/39084, but ended up not needing it. It might be better to just do away with `CoreTransformationContext` entirely, and just have any customized `TransformationContext` impelementations just patch in unsupported methods from `nullTransformationContext`.

Reply:
`nullTransformationContext` throw exceptions for some of the methods:

```ts 
{
    getEmitResolver: notImplemented,
    getEmitHost: notImplemented,
    getEmitHelperFactory: notImplemented,
    //....
 }
 ```

My problem is that this is not exactly safe. There are no guarantees that those methods will not get called. 

This is especially problematic now in the declaration transform, as this will contain both paths that rely on `getEmitHost` and `getEmitResolver` (the regular TSC declaration emit) but also paths that should not call these (when called from `transformaDeclaration`). I would like to make sure that it's not possible to call mistakenly depend on emitHost and emitResolver if they are not present.